tools/webpack/shared.js

```diff
@@ -25,7 +25,7 @@ const baseConfig = {
 				parallel: true,
 				terserOptions: {
 					output: {
-						comments: /translators:/i,
+						comments: /(translators:|wp:polyfill)/i,
```

Comment:
I think you could avoid having to preserve the comment through Terser by checking for the "wp:polyfill" comments in the `PROCESS_ASSETS_STAGE_OPTIMIZE_COMPATIBILITY` stage (rather than `PROCESS_ASSETS_STAGE_ANALYSE` as now).

You could pass the "needs polyfill" flag between the two hooks with a map of some sort or by doing something like `compilation.updateAsset( filename, v => v, { wpPolyfill: flag } );` to set it and then `asset.info.wpPolyfill` to check it.

Reply:
That would work. It would be a bit more complex, as it would require scanning all the chunks twice. We decided to keep the comment so we don't have to recalculate the hash generated for chunks. It sounds like a good follow up task.

Reply:
> It sounds like a good follow up task.

I'll take that as an invitation. 🙂

https://github.com/WordPress/gutenberg/pull/65582

Reply:
Awesome, I will have a look when I find some time in the next few days.